source/lib.ts

```diff
@@ -144,13 +144,26 @@ const parseOptions = (passedOptions: Partial<Options>): Configuration => {
 
 			return request.ip
 		},
-		handler(
-			_request: Request,
+		async handler(
+			request: Request,
 			response: Response,
 			_next: NextFunction,
 			_optionsUsed: Options,
-		): void {
-			response.status(config.statusCode).send(config.message)
+		): Promise<void> {
+			// Set the response status code
+			response.status(config.statusCode)
+			// Call the `message` if it is a function.
+			const message: unknown =
+				typeof config.message === 'function'
+					? await (
+							config.message as (request: Request, response: Response) => any
+					  )(request, response)
+					: config.message
+
+			// Send the response if writable.
+			if (!response.writableEnded) {
+				response.send(message || 'Too many requests, please try again later.')
```

Comment:
Good call putting a default value in here.

Reply:
Yes, that is a good catch! Although, could you please use the `??` operator here instead of the `||` operator?